baricadr/model/repos.py

```diff
@@ -157,42 +168,65 @@ def _get_freezable(self, path, remote_list, force=False):
                 if fnmatch.fnmatch(path, ex.strip()):
                     current_app.logger.info("Found excluded path: %s with expression %s" % (path, ex.strip()))
                     return
-            if (force or self._can_freeze(path)) and (self.relative_path(path) in remote_list):
+            if self._can_freeze(path, remote_list, force):
                 freezables.append(path)
         else:
             for root, subdirs, files in os.walk(path):
                 for name in files:
                     candidate = os.path.join(root, name)
-                    current_app.logger.info("Evaluating freezable for path: %s -> force %s can_freeze %s in remote_list %s" % (candidate, force, self._can_freeze(path), (self.relative_path(path) in remote_list)))
+                    current_app.logger.info("Evaluating freezable for path: %s " % (candidate))
                     excluded = False
                     for ex in excludes:
                         if fnmatch.fnmatch(candidate, ex.strip()):
                             current_app.logger.info("Found excluded path: %s with expression %s" % (candidate, ex.strip()))
                             excluded = True
                             break
-                    if not excluded and (force or self._can_freeze(candidate)) and (self.relative_path(candidate) in remote_list):
+                    if not excluded and self._can_freeze(candidate, remote_list, force):
                         freezables.append(candidate)
 
         return freezables
 
-    def _can_freeze(self, file_to_check):
+    def _can_freeze(self, file_to_check, remote_list, force):
         """
         Check if a file should be freezed or not
 
-        :type path: str
-        :param path: Path of a file to check
+        :type file_to_check: str
+        :param file_to_check: Path of a file to check
+
+        :type force: bool
+        :param force: Whether to ignore atime
 
         :rtype: bool
         :return: True if the file should be freezed
         """
 
+        relative_path = self.relative_path(file_to_check)
+        # Check if in remote list
+        remote_file = next((item for item in remote_list if item["Path"] == relative_path), None)
+
+        if not remote_file:
+            return False
+
+        # Check if modified since pulled
+        tz = get_localzone()
+
+        last_modif_remote = dateutil.parser.isoparse(remote_file['ModTime'])
+        last_modif_local = datetime.datetime.fromtimestamp(os.stat(file_to_check).st_mtime, tz=tz)
+
+        if force:
```

Comment:
shouldn't it be `if not force:`?

Reply:
I think we can remove the test and just log the value since it's relevant info. I just added the test to avoid polluting the log files (and just print the log for this test in particular)

Reply:
:+1: